AssertionError in aeppl.logp breaks sampling in pymc

## Description of your problem or feature request

**Please provide a minimal, self-contained, and reproducible example.**
https://github.com/aesara-devs/aeppl/blob/18275e2bba83f21da5cde256bd67d86fa5b32451/aeppl/logprob.py#L92

**Please provide the full traceback of any errors.**
```python
AssertionError: sigma > 0
Apply node that caused the error: Assert{msg='sigma > 0'}(Elemwise{Composite{((i0 + (i1 * sqr(i2))) - log(i3))}}[(0, 2)].0, All.0)
```

**Please provide any additional information below.**
Sampling with PyMC (master) fails with assertion errors and does not treat AssertionErorr as divergent sample

**Expected Behaviour**
```python
return -inf
```
**Possible Solution**
* provide a graph rewrite for fixing assertions
* replace Assert with switch

This snippet solved my issue
```python
aesara.assert_op.Assert = lambda name: (lambda res, *cond: aesara.tensor.switch(
    aesara.tensor.all(aesara.tensor.stack([c.all() for c in cond])), 
    res, 
    -np.inf
))
```
## Versions and main components

* Aesara version:
* Aesara config (`python -c "import aesara; print(aesara.config)"`)
* Python version:
* Operating system:
* How did you install Aesara: (conda/pip)

--- a/aeppl/logprob.py
+++ b/aeppl/logprob.py
@@ -4,12 +4,30 @@ from typing import Tuple
 import aesara.tensor as at
 import aesara.tensor.random.basic as arb
 import numpy as np
-from aesara.assert_op import Assert
 from aesara.graph.op import Op
+from aesara.raise_op import CheckAndRaise
 from aesara.tensor.slinalg import Cholesky, solve_lower_triangular
 from aesara.tensor.var import TensorVariable
 
 # from aesara.tensor.xlogx import xlogy0
+
+
+class ParameterValueError(ValueError):
+    """Exception for invalid parameters values in logprob graphs"""
+
+
+class CheckParameterValue(CheckAndRaise):
+    """Implements a parameter value check in a logprob graph.
+
+    Raises `ParameterValueError` if the check is not True.
+    """
+
+    def __init__(self, msg=""):
+        super().__init__(ParameterValueError, msg)
+
+    def __str__(self):
+        return f"Check{{{self.msg}}}"
+
 
 cholesky = Cholesky(lower=True, on_error="nan")
 
@@ -108,7 +126,7 @@ def uniform_logcdf(op, value, *inputs, **kwargs):
         ),
     )
 
-    res = Assert("lower <= upper")(res, at.all(at.le(lower, upper)))
+    res = CheckParameterValue("lower <= upper")(res, at.all(at.le(lower, upper)))
     return res
 
 
@@ -121,7 +139,7 @@ def normal_logprob(op, values, *inputs, **kwargs):
         - at.log(at.sqrt(2.0 * np.pi))
         - at.log(sigma)
     )
-    res = Assert("sigma > 0")(res, at.all(at.gt(sigma, 0.0)))
+    res = CheckParameterValue("sigma > 0")(res, at.all(at.gt(sigma, 0.0)))
     return res
 
 
@@ -136,7 +154,7 @@ def normal_logcdf(op, value, *inputs, **kwargs):
         at.log1p(-at.erfc(z / at.sqrt(2.0)) / 2.0),
     )
 
-    res = Assert("sigma > 0")(res, at.all(at.gt(sigma, 0.0)))
+    res = CheckParameterValue("sigma > 0")(res, at.all(at.gt(sigma, 0.0)))
     return res
 
 
@@ -150,7 +168,7 @@ def halfnormal_logprob(op, values, *inputs, **kwargs):
         - at.log(sigma)
     )
     res = at.switch(at.ge(value, loc), res, -np.inf)
-    res = Assert("sigma > 0")(res, at.all(at.gt(sigma, 0.0)))
+    res = CheckParameterValue("sigma > 0")(res, at.all(at.gt(sigma, 0.0)))
     return res
 
 
@@ -164,7 +182,7 @@ def beta_logprob(op, values, *inputs, **kwargs):
         - (at.gammaln(alpha) + at.gammaln(beta) - at.gammaln(alpha + beta))
     )
     res = at.switch(at.bitwise_and(at.ge(value, 0.0), at.le(value, 1.0)), res, -np.inf)
-    res = Assert("0 <= value <= 1, alpha > 0, beta > 0")(
+    res = CheckParameterValue("0 <= value <= 1, alpha > 0, beta > 0")(
         res, at.all(at.gt(alpha, 0.0)), at.all(at.gt(beta, 0.0))
     )
     return res
@@ -176,7 +194,7 @@ def exponential_logprob(op, values, *inputs, **kwargs):
     (mu,) = inputs[3:]
     res = -at.log(mu) - value / mu
     res = at.switch(at.ge(value, 0.0), res, -np.inf)
-    res = Assert("mu > 0")(res, at.all(at.gt(mu, 0.0)))
+    res = CheckParameterValue("mu > 0")(res, at.all(at.gt(mu, 0.0)))
     return res
 
 
@@ -198,7 +216,7 @@ def lognormal_logprob(op, values, *inputs, **kwargs):
         - at.log(value)
     )
     res = at.switch(at.gt(value, 0.0), res, -np.inf)
-    res = Assert("sigma > 0")(res, at.all(at.gt(sigma, 0)))
+    res = CheckParameterValue("sigma > 0")(res, at.all(at.gt(sigma, 0)))
     return res
 
 
@@ -208,7 +226,7 @@ def pareto_logprob(op, values, *inputs, **kwargs):
     alpha, m = inputs[3:]
     res = at.log(alpha) + xlogy0(alpha, m) - xlogy0(alpha + 1.0, value)
     res = at.switch(at.ge(value, m), res, -np.inf)
-    res = Assert("alpha > 0, m > 0")(
+    res = CheckParameterValue("alpha > 0, m > 0")(
         res, at.all(at.gt(alpha, 0.0)), at.all(at.gt(m, 0.0))
     )
     return res
@@ -219,7 +237,7 @@ def cauchy_logprob(op, values, *inputs, **kwargs):
     (value,) = values
     alpha, beta = inputs[3:]
     res = -at.log(np.pi) - at.log(beta) - at.log1p(at.pow((value - alpha) / beta, 2))
-    res = Assert("beta > 0")(res, at.all(at.gt(beta, 0.0)))
+    res = CheckParameterValue("beta > 0")(res, at.all(at.gt(beta, 0.0)))
     return res
 
 
@@ -244,7 +262,7 @@ def gamma_logprob(op, values, *inputs, **kwargs):
         + xlogy0(alpha - 1, value)
     )
     res = at.switch(at.ge(value, 0.0), res, -np.inf)
-    res = Assert("alpha > 0, beta > 0")(
+    res = CheckParameterValue("alpha > 0, beta > 0")(
         res, at.all(at.gt(alpha, 0.0)), at.all(at.gt(beta, 0.0))
     )
     return res
@@ -262,7 +280,7 @@ def invgamma_logprob(op, values, *inputs, **kwargs):
         + xlogy0(-alpha - 1, value)
     )
     res = at.switch(at.ge(value, 0.0), res, -np.inf)
-    res = Assert("alpha > 0, beta > 0")(
+    res = CheckParameterValue("alpha > 0, beta > 0")(
         res, at.all(at.gt(alpha, 0.0)), at.all(at.gt(beta, 0.0))
     )
     return res
@@ -287,7 +305,7 @@ def wald_logprob(op, values, *inputs, **kwargs):
     )
 
     res = at.switch(at.gt(value, 0.0), res, -np.inf)
-    res = Assert("mu > 0, scale > 0")(
+    res = CheckParameterValue("mu > 0, scale > 0")(
         res, at.all(at.gt(mu, 0.0)), at.all(at.gt(scale, 0.0))
     )
     return res
@@ -304,7 +322,7 @@ def weibull_logprob(op, values, *inputs, **kwargs):
         - at.pow(value / beta, alpha)
     )
     res = at.switch(at.ge(value, 0.0), res, -np.inf)
-    res = Assert("alpha > 0, beta > 0")(
+    res = CheckParameterValue("alpha > 0, beta > 0")(
         res, at.all(at.gt(alpha, 0.0)), at.all(at.gt(beta, 0.0))
     )
     return res
@@ -318,7 +336,7 @@ def vonmises_logprob(op, values, *inputs, **kwargs):
     res = at.switch(
         at.bitwise_and(at.ge(value, -np.pi), at.le(value, np.pi)), res, -np.inf
     )
-    res = Assert("kappa > 0")(res, at.all(at.gt(kappa, 0.0)))
+    res = CheckParameterValue("kappa > 0")(res, at.all(at.gt(kappa, 0.0)))
     return res
 
 
@@ -334,7 +352,7 @@ def triangular_logprob(op, values, *inputs, **kwargs):
     res = at.switch(
         at.bitwise_and(at.le(lower, value), at.le(value, upper)), res, -np.inf
     )
-    res = Assert("lower <= c, c <= upper")(
+    res = CheckParameterValue("lower <= c, c <= upper")(
         res, at.all(at.le(lower, c)), at.all(at.le(c, upper))
     )
     return res
@@ -346,7 +364,7 @@ def gumbel_logprob(op, values, *inputs, **kwargs):
     mu, beta = inputs[3:]
     z = (value - mu) / beta
     res = -z - at.exp(-z) - at.log(beta)
-    res = Assert("0 < beta")(res, at.all(at.lt(0.0, beta)))
+    res = CheckParameterValue("0 < beta")(res, at.all(at.lt(0.0, beta)))
     return res
 
 
@@ -356,7 +374,7 @@ def logistic_logprob(op, values, *inputs, **kwargs):
     mu, s = inputs[3:]
     z = (value - mu) / s
     res = -z - at.log(s) - 2.0 * at.log1p(at.exp(-z))
-    res = Assert("0 < s")(res, at.all(at.lt(0.0, s)))
+    res = CheckParameterValue("0 < s")(res, at.all(at.lt(0.0, s)))
     return res
 
 
@@ -366,7 +384,9 @@ def binomial_logprob(op, values, *inputs, **kwargs):
     n, p = inputs[3:]
     res = binomln(n, value) + xlogy0(value, p) + xlogy0(n - value, 1.0 - p)
     res = at.switch(at.bitwise_and(at.le(0, value), at.le(value, n)), res, -np.inf)
-    res = Assert("0 <= p, p <= 1")(res, at.all(at.le(0.0, p)), at.all(at.le(p, 1.0)))
+    res = CheckParameterValue("0 <= p, p <= 1")(
+        res, at.all(at.le(0.0, p)), at.all(at.le(p, 1.0))
+    )
     return res
 
 
@@ -380,7 +400,7 @@ def betabinomial_logprob(op, values, *inputs, **kwargs):
         - betaln(alpha, beta)
     )
     res = at.switch(at.bitwise_and(at.le(0, value), at.le(value, n)), res, -np.inf)
-    res = Assert("0 < alpha, 0 < beta")(
+    res = CheckParameterValue("0 < alpha, 0 < beta")(
         res, at.all(at.lt(0.0, alpha)), at.all(at.lt(0.0, beta))
     )
     return res
@@ -392,7 +412,9 @@ def bernoulli_logprob(op, values, *inputs, **kwargs):
     (p,) = inputs[3:]
     res = at.switch(value, at.log(p), at.log(1.0 - p))
     res = at.switch(at.bitwise_and(at.le(0, value), at.le(value, 1)), res, -np.inf)
-    res = Assert("0 <= p <= 1")(res, at.all(at.le(0.0, p)), at.all(at.le(p, 1.0)))
+    res = CheckParameterValue("0 <= p <= 1")(
+        res, at.all(at.le(0.0, p)), at.all(at.le(p, 1.0))
+    )
     return res
 
 
@@ -402,7 +424,7 @@ def poisson_logprob(op, values, *inputs, **kwargs):
     (mu,) = inputs[3:]
     res = xlogy0(value, mu) - at.gammaln(value + 1) - mu
     res = at.switch(at.le(0, value), res, -np.inf)
-    res = Assert("0 <= mu")(res, at.all(at.le(0.0, mu)))
+    res = CheckParameterValue("0 <= mu")(res, at.all(at.le(0.0, mu)))
     res = at.switch(at.bitwise_and(at.eq(mu, 0.0), at.eq(value, 0.0)), 0.0, res)
     return res
 
@@ -413,7 +435,7 @@ def poisson_logcdf(op, value, *inputs, **kwargs):
     value = at.floor(value)
     res = at.log(at.gammaincc(value + 1, mu))
     res = at.switch(at.le(0, value), res, -np.inf)
-    res = Assert("0 <= mu")(res, at.all(at.le(0.0, mu)))
+    res = CheckParameterValue("0 <= mu")(res, at.all(at.le(0.0, mu)))
     return res
 
 
@@ -428,7 +450,9 @@ def nbinom_logprob(op, values, *inputs, **kwargs):
         + xlogy0(n, n / (mu + n))
     )
     res = at.switch(at.le(0, value), res, -np.inf)
-    res = Assert("0 < mu, 0 < n")(res, at.all(at.lt(0.0, mu)), at.all(at.lt(0.0, n)))
+    res = CheckParameterValue("0 < mu, 0 < n")(
+        res, at.all(at.lt(0.0, mu)), at.all(at.lt(0.0, n))
+    )
     res = at.switch(at.gt(n, 1e10), poisson_logprob(op, values, *inputs[:3], mu), res)
     return res
 
@@ -439,7 +463,9 @@ def geometric_logprob(op, values, *inputs, **kwargs):
     (p,) = inputs[3:]
     res = at.log(p) + xlogy0(value - 1, 1 - p)
     res = at.switch(at.le(1, value), res, -np.inf)
-    res = Assert("0 <= p <= 1")(res, at.all(at.le(0.0, p)), at.all(at.ge(1.0, p)))
+    res = CheckParameterValue("0 <= p <= 1")(
+        res, at.all(at.le(0.0, p)), at.all(at.ge(1.0, p))
+    )
     return res
 
 
@@ -492,7 +518,9 @@ def categorical_logprob(op, values, *inputs, **kwargs):
     res = at.switch(
         at.bitwise_and(at.le(0, value), at.lt(value, at.shape(p)[-1])), res, -np.inf
     )
-    res = Assert("0 <= p <= 1")(res, at.all(at.ge(p, 0.0)), at.all(at.le(p, 1.0)))
+    res = CheckParameterValue("0 <= p <= 1")(
+        res, at.all(at.ge(p, 0.0)), at.all(at.le(p, 1.0))
+    )
     return res
 
 
@@ -525,7 +553,7 @@ def mvnormal_logprob(op, values, *inputs, **kwargs):
 
     n = value.shape[-1]
     res = -0.5 * n * at.log(2 * np.pi) - 0.5 * quaddist - logdet
-    res = Assert("0 < diag(Sigma)")(res, all_pos_definite)
+    res = CheckParameterValue("0 < diag(Sigma)")(res, all_pos_definite)
     return res
 
 
@@ -543,7 +571,7 @@ def dirichlet_logprob(op, values, *inputs, **kwargs):
         res,
         -np.inf,
     )
-    res = Assert("0 < alpha")(res, at.all(at.lt(0.0, alpha)))
+    res = CheckParameterValue("0 < alpha")(res, at.all(at.lt(0.0, alpha)))
     return res
 
 
@@ -559,7 +587,7 @@ def multinomial_logprob(op, values, *inputs, **kwargs):
         res,
         -np.inf,
     )
-    res = Assert("p <= 1, sum(p) == 1, n >= 0")(
+    res = CheckParameterValue("p <= 1, sum(p) == 1, n >= 0")(
         res,
         at.all(at.le(p, 1)),
         at.all(at.eq(at.sum(p, axis=-1), 1)),

--- a/aeppl/truncation.py
+++ b/aeppl/truncation.py
@@ -3,7 +3,6 @@ from typing import List, Optional
 
 import aesara.tensor as at
 import numpy as np
-from aesara.assert_op import Assert
 from aesara.graph.basic import Node
 from aesara.graph.fg import FunctionGraph
 from aesara.graph.opt import local_optimizer
@@ -13,7 +12,7 @@ from aesara.tensor.elemwise import Elemwise
 from aesara.tensor.var import TensorConstant
 
 from aeppl.abstract import MeasurableVariable, assign_custom_measurable_outputs
-from aeppl.logprob import _logcdf, _logprob
+from aeppl.logprob import CheckParameterValue, _logcdf, _logprob
 from aeppl.opt import rv_sinking_db
 
 
@@ -119,7 +118,7 @@ def censor_logprob(op, values, base_rv, lower_bound, upper_bound, **kwargs):
         )
 
     if is_lower_bounded and is_upper_bounded:
-        logprob = Assert("lower_bound <= upper_bound")(
+        logprob = CheckParameterValue("lower_bound <= upper_bound")(
             logprob, at.all(at.le(lower_bound, upper_bound))
         )
 

--- a/setup.py
+++ b/setup.py
@@ -17,7 +17,7 @@ setup(
     install_requires=[
         "numpy>=1.18.1",
         "scipy>=1.4.0",
-        "aesara >= 2.2.1",
+        "aesara >= 2.3.1",
     ],
     tests_require=["pytest"],
     long_description=open("README.rst").read() if exists("README.rst") else "",

--- a/tests/test_logprob.py
+++ b/tests/test_logprob.py
@@ -4,8 +4,9 @@ import aesara.tensor as at
 import numpy as np
 import pytest
 import scipy.stats as stats
+from aesara import function
 
-from aeppl.logprob import logcdf, logprob
+from aeppl.logprob import ParameterValueError, logcdf, logprob
 
 # @pytest.fixture(scope="module", autouse=True)
 # def set_aesara_flags():
@@ -197,7 +198,7 @@ def test_exponential_logprob(dist_params, obs, size, error):
 
     x = at.random.exponential(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, mu):
         return stats.expon.logpdf(obs, scale=mu)
@@ -244,7 +245,7 @@ def test_lognormal_logprob(dist_params, obs, size, error):
 
     x = at.random.lognormal(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, mu, sigma):
         return stats.lognorm.logpdf(obs, s=sigma, scale=np.exp(mu))
@@ -269,7 +270,7 @@ def test_pareto_logprob(dist_params, obs, size, error):
 
     x = at.random.pareto(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, b, scale):
         return stats.pareto.logpdf(obs, b, scale=scale)
@@ -294,7 +295,7 @@ def test_halfcauchy_logprob(dist_params, obs, size, error):
 
     x = at.random.halfcauchy(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, alpha, beta):
         return stats.halfcauchy.logpdf(obs, loc=alpha, scale=beta)
@@ -319,7 +320,7 @@ def test_gamma_logprob(dist_params, obs, size, error):
 
     x = at.random.gamma(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, alpha, inv_beta):
         return stats.gamma.logpdf(obs, alpha, scale=1.0 / inv_beta)
@@ -344,7 +345,7 @@ def test_invgamma_logprob(dist_params, obs, size, error):
 
     x = at.random.invgamma(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, alpha, beta):
         return stats.invgamma.logpdf(obs, alpha, scale=beta)
@@ -371,7 +372,7 @@ def test_chisquare_logprob(dist_params, obs, size, error):
 
     x = at.random.chisquare(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, df):
         return stats.chi2.logpdf(obs, df)
@@ -396,7 +397,7 @@ def test_wald_logprob(dist_params, obs, size, error):
 
     x = at.random.wald(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, mean, scale):
         return stats.invgauss.logpdf(obs, mean / scale, scale=scale)
@@ -421,7 +422,7 @@ def test_weibull_logprob(dist_params, obs, size, error):
 
     x = at.random.weibull(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, alpha, beta):
         return stats.weibull_min.logpdf(obs, alpha, scale=beta)
@@ -456,7 +457,7 @@ def test_vonmises_logprob(dist_params, obs, size, error):
 
     x = at.random.vonmises(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, mu, kappa):
         return stats.vonmises.logpdf(obs, kappa, loc=mu)
@@ -486,7 +487,7 @@ def test_triangular_logprob(dist_params, obs, size, error):
 
     x = at.random.triangular(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, lower, mode, upper):
         return stats.triang.logpdf(
@@ -513,7 +514,7 @@ def test_gumbel_logprob(dist_params, obs, size, error):
 
     x = at.random.gumbel(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, mu, beta):
         return stats.gumbel_r.logpdf(obs, loc=mu, scale=beta)
@@ -538,7 +539,7 @@ def test_logistic_logprob(dist_params, obs, size, error):
 
     x = at.random.logistic(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, mu, s):
         return stats.logistic.logpdf(obs, loc=mu, scale=s)
@@ -569,7 +570,7 @@ def test_binomial_logprob(dist_params, obs, size, error):
 
     x = at.random.binomial(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, n, p):
         return stats.binom.logpmf(obs, n, p)
@@ -599,7 +600,7 @@ def test_betabinomial_logprob(dist_params, obs, size, error):
 
     x = at.random.betabinom(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, n, alpha, beta):
         return stats.betabinom.logpmf(obs, n, alpha, beta)
@@ -624,7 +625,7 @@ def test_bernoulli_logprob(dist_params, obs, size, error):
 
     x = at.random.bernoulli(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, p):
         return stats.bernoulli.logpmf(obs, p)
@@ -654,7 +655,7 @@ def test_poisson_logprob(dist_params, obs, size, error):
 
     x = at.random.poisson(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, mu):
         return stats.poisson.logpmf(obs, mu)
@@ -684,7 +685,7 @@ def test_poisson_logcdf(dist_params, obs, size, error):
 
     x = at.random.poisson(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logcdf(obs, mu):
         return stats.poisson.logcdf(obs, mu)
@@ -722,7 +723,7 @@ def test_nbinom_logprob(dist_params, obs, size, error):
 
     x = at.random.nbinom(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, n, p):
         return stats.nbinom.logpmf(obs, n, p)
@@ -752,7 +753,7 @@ def test_geometric_logprob(dist_params, obs, size, error):
 
     x = at.random.geometric(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, p):
         return stats.geom.logpmf(obs, p)
@@ -801,7 +802,7 @@ def test_hypergeometric_logprob(dist_params, obs, size, error):
             (np.array([-0.5, 0.5]),),
             np.array([0, 1], dtype=np.int64),
             (),
-            AssertionError,
+            ParameterValueError,
             True,
         ),
         (
@@ -933,7 +934,7 @@ def test_mvnormal_logprob(dist_params, obs, size, error):
 
     x = at.random.multivariate_normal(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, mu, cov):
         return stats.multivariate_normal.logpdf(obs, mu, cov)
@@ -966,7 +967,7 @@ def test_dirichlet_logprob(dist_params, obs, size, error):
 
     x = at.random.dirichlet(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, alpha):
         return stats.dirichlet.logpdf(obs.T, alpha)
@@ -999,10 +1000,22 @@ def test_multinomial_logprob(dist_params, obs, size, error):
 
     x = at.random.multinomial(*dist_params_at, size=size_at)
 
-    cm = contextlib.suppress() if not error else pytest.raises(AssertionError)
+    cm = contextlib.suppress() if not error else pytest.raises(ParameterValueError)
 
     def scipy_logprob(obs, n, p):
         return stats.multinomial.logpmf(obs, n, p)
 
     with cm:
         scipy_logprob_tester(x, obs, dist_params, test_fn=scipy_logprob)
+
+
+def test_CheckParameter():
+    mu = at.constant(0)
+    sigma = at.scalar("sigma")
+    x_rv = at.random.normal(mu, sigma, name="x")
+    x_vv = at.constant(0)
+    x_logp = logprob(x_rv, x_vv)
+
+    x_logp_fn = function([sigma], x_logp)
+    with pytest.raises(ParameterValueError, match="sigma > 0"):
+        x_logp_fn(-1)

